### ISSUE TITLE
(Dev) After send money it will show "$0.00" in preview card

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#18881]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open the app
2. Go to any chat > Send any amount
3. You'll notice "$0.00" in preview card

## Expected Result:
It should should the correct amount

## Actual Result:
It is showing "$0.00"

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** DEV
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:** needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
![image](https://github.com/Expensify/App/assets/43996225/083ff30c-97ed-43cf-ab7b-e4cd8aefba11)

**Expensify/Expensify Issue URL:**
**Issue reported by:** @daraksha-dk 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1683903988041489

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by kushu7** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
After sending money its showing $0.00 in preview card
### What is the root cause of that problem?
When report action `type` of payment is `pay`. we remove `amount, currency, comment` from originalMessage.
https://github.com/Expensify/App/blob/293f23b57be2bca3f3eb0af903d5e0c2894cb36a/src/libs/ReportUtils.js#L1327-L1334

We save it in `originalMessage.IOUDetails`.
### What changes do you think we should make in order to solve the problem?

For resolving this we can do the following. when action type is `pay` we can read data from `originalMessage.IOUDetails`.

    // We store amount, comment, currency in IOUDetails when type = pay
    const prefix = lodashGet(props.action, 'originalMessage.type', null) === CONST.IOU.REPORT_ACTION_TYPE.PAY ? 'originalMessage.IOUDetails' : 'originalMessage';

    // If props.action is undefined then we are displaying within IOUDetailsModal and should use the full report amount
    const requestAmount = props.isIOUAction ? lodashGet(props.action, `${prefix}.amount`, 0) : ReportUtils.getMoneyRequestTotal(props.iouReport);
    const requestCurrency = props.isIOUAction ? lodashGet(props.action, `${prefix}.currency`, CONST.CURRENCY.USD) : props.iouReport.currency;
    const requestComment = lodashGet(props.action, `${prefix}.comment`, '');
### What alternative solutions did you explore? (Optional)
None

**Comment by trjExpensify** [comment]
Coming from [the thread](https://expensify.slack.com/archives/C049HHMV9SM/p1683916867891729?thread_ts=1683903988.041489&cid=C049HHMV9SM), @mollfpr is this due to your [currency PR](https://github.com/Expensify/App/pull/18783) perhaps? 

Also, this is reproducible on staging, so I've updated the OP.

**Comment by mollfpr** [comment]
@trjExpensify Probably yes. I’ll check it in the morning.

**Comment by trjExpensify** [comment]
Sounds good! Can I co-assign you in the meantime?

**Comment by mollfpr** [comment]
Yup,

**Comment by trjExpensify** [comment]
Awesome, thanks! 

CC: @JmillsExpensify @luacmartins @Julesssss @mountiny

**Comment by luacmartins** [comment]
@mollfpr I think the issue is in [this line](https://github.com/Expensify/App/blob/d3d17c1a2a51d021cf33b869304f415c2c6e1064/src/libs/ReportUtils.js#L873), since after settling the report, there's no more `outstandingIOU` and we just display 0.

**Comment by mollfpr** [comment]
@luacmartins Yes! I guess we don't need that and check with `isMoneyRequest` since it has type `iou`.

![Screenshot 2023-05-13 at 21 59 42](https://github.com/Expensify/App/assets/25520267/fcd8470f-e23a-434b-9ec9-6ac47d884c1c)

![Screenshot 2023-05-13 at 21 58 34](https://github.com/Expensify/App/assets/25520267/4cdd6fd5-b01d-41b7-a3f4-a7b8ec7a6ca3)

**Comment by mollfpr** [comment]
@luacmartins is not an issue with `getMoneyRequestTotal`.

We have this `isIOUAction` default to true.

https://github.com/Expensify/App/blob/293f23b57be2bca3f3eb0af903d5e0c2894cb36a/src/components/ReportActionItem/IOUPreview.js#L120

https://github.com/Expensify/App/blob/293f23b57be2bca3f3eb0af903d5e0c2894cb36a/src/components/ReportActionItem/IOUPreview.js#L150

So the amount to display will be from `action.originalMessage.amount`, but as you can see, the `props.action` passed is missing the `amount` value.

![Screenshot 2023-05-13 at 22 13 21](https://github.com/Expensify/App/assets/25520267/2750fbaa-b8fd-435a-aea2-8ad5fe54b506)

**Comment by mollfpr** [comment]
We have different `action.originalMessage` object structures between _**Send Money**_, _**Request Money**_, and **Request Money Paid**.

<details>
<summary>Send money response</summary>


![Send Money Pusher Response](https://github.com/Expensify/App/assets/25520267/8140bce2-510c-45f9-bc8e-f5708ee495b1)
</details>


<details>
<summary>Request money response</summary>

![Request Money Pusher Response](https://github.com/Expensify/App/assets/25520267/77888579-e37c-4298-8840-1d056d171462)

</details>

<details>
<summary>Request money paid response</summary>

![Request Money Payed Pusher Response](https://github.com/Expensify/App/assets/25520267/aacb33d1-56d0-4d8d-afc4-aff2218311d7)

</details>


The easiest solution I can think is to use the full report object rather than the action object.
```diff
-    const requestAmount = props.isIOUAction ? lodashGet(props.action, 'originalMessage.amount', 0) : ReportUtils.getMoneyRequestTotal(props.iouReport);
-    const requestCurrency = props.isIOUAction ? lodashGet(props.action, 'originalMessage.currency', CONST.CURRENCY.USD) : props.iouReport.currency;
+    const requestAmount = ReportUtils.getMoneyRequestTotal(props.iouReport);
+    const requestCurrency = props.iouReport.currency;
```

We already use the complete report total to determine whether we should display the IOU preview.
https://github.com/Expensify/App/blob/293f23b57be2bca3f3eb0af903d5e0c2894cb36a/src/components/ReportActionItem/IOUPreview.js#L130-L136

**Comment by mountiny** [comment]
@mollfpr I can confirm all of those `originalMessage` formats are expected so feel free to update the App to handle this better

**Comment by mollfpr** [comment]
@trjExpensify @mountiny @luacmartins I open a draft [PR](https://github.com/Expensify/App/pull/18902).

I also noticed another bug with `IOUPreview` on the settled request money; there's a space because the action doesn't have the `comment.

![Android bug](https://github.com/Expensify/App/assets/25520267/53e97c78-d7be-404a-8a9a-d926b8d69676)

![Screenshot 2023-05-14 at 12 58 32](https://github.com/Expensify/App/assets/25520267/9fd3ac43-0500-43c5-baa7-214c797d2260)

**Comment by trjExpensify** [comment]
Yeah, we have a PR for it that needs approving. https://github.com/Expensify/App/pull/18774

**Comment by mountiny** [comment]
Merged and CPing

**Comment by trjExpensify** [comment]
Added the `external` label to get this job created on Upwork. Offers sent for the following: 

$1,500 - @mollfpr for the fix + #urgency bonus 
$250 - @daraksha-dk for the bug report. 

Let me know when you accept and I'll settle the contracts. 👍

**Comment by trjExpensify** [comment]
Settled up with @mollfpr 👍

**Comment by trjExpensify** [comment]
Excellent. Settled up, thanks everyone!
